### PR TITLE
Greg

### PR DESCRIPTION
**Pull request by GregoryElliott**

### CODE REVIEW

File: Assets/scripts/MelodyUnit.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+using System.Linq;
+using System.Collections;
+
+public class MelodyUnit : Unit {
+    public Color LeadingColor;
+    public override void Initialize()
+    {
+        base.Initialize();
+        transform.position += new Vector3(0, 0, -1);
+        GetComponent<Renderer>().material.color = LeadingColor;
```
Comment (armsnyder) [inline Assets/scripts/MelodyUnit.cs:13]: Our units won't have renderers / materials since they're 2D sprites. Are we considering going 3D?

**Comment by armsnyder** [comment]
It looks good. I'm not as familiar with the framework as you, so I can't comment on the unit / cell grid spawners.

**Comment by cegrief** [comment]
I think we might not actually need some of these classes, and will probably have to make some changes in how the board is generated in general if we are doing things like terrain type or whatever, but for now this seems fine.

**Comment by armsnyder** [comment]
Do you want to merge it or should I lol

**Pull request merged**